lib/ipam/ipam_block.go

```diff
@@ -416,15 +416,14 @@ func incrementIP(ip cnet.IP, increment *big.Int) cnet.IP {
 	return intToIP(sum)
 }
 
-func ipToOrdinal(ip cnet.IP, b allocationBlock) int {
+func ipToOrdinal(ip cnet.IP, b allocationBlock) (int, error) {
 	ip_int := ipToInt(ip)
 	base_int := ipToInt(cnet.IP{b.CIDR.IP})
 	ord := big.NewInt(0).Sub(ip_int, base_int).Int64()
 	if ord < 0 || ord >= blockSize {
-		// IP address not in the given block.
-		log.Panicf("IP %s not in block %s", ip, b.CIDR)
+		return 0, errors.New(fmt.Sprintf("IP %s not in block %s", ip, b.CIDR))
```

Comment:
@caseydavenport (or @gunjan5 ? ) What would you do when you can't return nil as an integer here? Convert to a struct or pointer?

Reply:
returning the zero value is fine, I think, if you're also returning the error with it. Calling code should be checking the error before using the int value

Reply:
also just use `fmt.Errorf("IP %s not in block %s", ip, b.CIDR)`

Reply:
I'd just return the zero value, like you're doing, and then have the calling code check errors.